troposphere/__init__.py

```diff
@@ -659,36 +772,36 @@ def _update(self, d, values):
             d[values.title] = values
         return values
 
-    def add_output(self, output):
+    def add_output(self, output: Output) -> Output:
         if len(self.outputs) >= MAX_OUTPUTS:
             raise ValueError("Maximum outputs %d reached" % MAX_OUTPUTS)
         return self._update(self.outputs, output)
 
-    def add_mapping(self, name, mapping):
+    def add_mapping(self, name: str, mapping: Dict[str, Any]) -> None:
         if len(self.mappings) >= MAX_MAPPINGS:
             raise ValueError("Maximum mappings %d reached" % MAX_MAPPINGS)
         if name not in self.mappings:
             self.mappings[name] = {}
         self.mappings[name].update(mapping)
 
-    def add_parameter(self, parameter):
+    def add_parameter(self, parameter: Parameter) -> Parameter:
         if len(self.parameters) >= MAX_PARAMETERS:
             raise ValueError("Maximum parameters %d reached" % MAX_PARAMETERS)
         return self._update(self.parameters, parameter)
 
-    def get_or_add_parameter(self, parameter):
+    def get_or_add_parameter(self, parameter: Parameter) -> Parameter:
         if parameter.title in self.parameters:
             return self.parameters[parameter.title]
         else:
             self.add_parameter(parameter)
         return parameter
 
-    def add_resource(self, resource):
+    def add_resource(self, resource: __BaseAWSObjectTypeVar) -> __BaseAWSObjectTypeVar:
```

Comment:
This does not match the implementation of `_update()` which accepts also a `list[__BaseAWSObjectTypeVar]`.  Same for `add_parameter` and `add_output`.

Was this intentional?